src/mono/mono/mini/aot-compiler.c

```diff
@@ -6589,6 +6589,7 @@ emit_and_reloc_code (MonoAotCompile *acfg, MonoMethod *method, guint8 *code, gui
 						direct_call_target = symbol;
 						patch_info->type = MONO_PATCH_INFO_NONE;
 					} else if ((m_class_get_image (patch_info->data.method->klass) == acfg->image) && !got_only && is_direct_callable (acfg, method, patch_info)) {
```

Comment:
Not sure if I understand this correctly, but should this: 
```
is_direct_callable (acfg, method, patch_info)
``` 
actually be:
```
is_direct_callable (acfg, cmethod, patch_info)
```
which might be causing issues for iOS (as the called method might require initialization)?

Reply:
Might help as it looks like static fields and class attributes of callee are "corrupted/not loaded properly" and ended up being NULL.